src/JWT.Extensions.AspNetCore/FailedTicketContext.cs

```diff
@@ -0,0 +1,21 @@
+namespace JWT.Extensions.AspNetCore;
+
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+public class FailedTicketContext
+{
+    public FailedTicketContext(ILogger logger, Exception exception, HttpContext context, JwtAuthenticationOptions options)
```

Comment:
Why do we need a separate class for the failed scenario?

Reply:
For this pattern, I'm copying what Microsoft has done with their events classes.

- https://github.com/dotnet/aspnetcore/blob/main/src/Security/Authentication/Cookies/src/CookieAuthenticationEvents.cs#L115
- https://github.com/dotnet/aspnetcore/blob/main/src/Security/Authentication/OAuth/src/Events/OAuthEvents.cs#L30
- https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/dev/src/AspNet.Security.OAuth.Okta/OktaAuthenticationHandler.cs#L57

Reply:
Sorry, I don't get it. I mean I get that you're following the pattern and it's the right thing to do. But the links don't include `FailedTicketContext`

Reply:
Oh, for sure! The biggest reason is that the FailedTicketContext has an Exception property, an the SuccessfulTicketContext has an AuthenticationTicket property. If I had to use one class to represent both, I would then need to make `Ticket` and `Exception` properties nullable. Dedicated classes allow each property to be non-null, which makes sense because in the happy path we ALWAYS have a Ticket and NEVER an error. And vice versa in the other path.

That is the biggest reason from a DX / API design stand point.

Really, the secondary reason is that its a pattern we see in the others.

The links don't include a `FailedTicketContext` since that is _OUR_ class. But for example:

[Cookie](https://github.com/dotnet/aspnetcore/blob/main/src/Security/Authentication/Cookies/src/CookieAuthenticationEvents.cs#L115)
- CookieValidatePrincipalContext
- CookieSlidingExpirationContext
- CookieSigningInContext
- CookieSignedInContext
- CookieSigningOutContext
- RedirectContext<CookieAuthenticationOptions>

[OAuth](https://github.com/dotnet/aspnetcore/blob/main/src/Security/Authentication/OAuth/src/Events/OAuthEvents.cs#L30)
- OAuthCreatingTicketContext
- RedirectContext<OAuthOptions>
- which inherits from [RemoteAuthenticationEvents](https://github.com/dotnet/aspnetcore/blob/main/src/Security/Authentication/Core/src/Events/RemoteAuthenticationEvents.cs#L9)
    - AccessDeniedContext
    - RemoteFailureContext
   - TicketReceivedContext

Since I see this pattern of very explicit context classes for each event, I chose to follow that path. I should probably go through and convert our other events to use context objects as well. 

I'm also interested in taking a pass through the Authentication Handles so we can also get AuthenticationProperties passed through as well. 

I hope that I addressed your question. :)

Reply:
Yes, it does. Thank you!